### ISSUE TITLE
HAProxyController sshd fails: requires root user for scp to /usr/local/etc/haproxy/haproxy.cfg

### ISSUE THREAD

**Issue by aledsage** [brooklyncentral/clocker#256]
Since the recent sshd container changes to Clocker (i.e. where we set up credentials, rather than leaving root with the default auto-generated password!), the `HAProxyController` entity fails.

This is because it tries to scp directly to `/usr/local/etc/haproxy/haproxy.cfg`, but that directory is owned by root so it fails.

A workaround for such docker containers would be to tell it to use root as the sshd user. (Better yet would be to scp to a temp directory, and then `sudo mv` the file).

For telling it to use `root` for sshd, I can't see how to get that injected. I tried the blueprint below, but it doesn't work:

```
name: haproxy @ clocker
location:
  my-docker-cloud:
    user: root
services:
- type: brooklyn.entity.proxy.haproxy.HAProxyController
  brooklyn.config:
    docker.image.name: haproxy
    docker.image.tag: 1.5.9
    install.dir: /usr/local/sbin/
    run.dir: /usr/local/etc/haproxy/
    member.sensor.hostname: "host.subnet.hostname"
    provisioning.properties:
      user: root
```

**Comment by aledsage** [comment]
This blueprint works:

```
name: haproxy @ clocker
location: my-docker-cloud
services:
- type: brooklyn.entity.proxy.haproxy.HAProxyController
  brooklyn.config:
    docker.image.name: haproxy
    docker.image.tag: 1.5.9
    install.dir: /usr/local/sbin/
    run.dir: /usr/local/etc/haproxy/
    member.sensor.hostname: "host.subnet.hostname"
    provisioning.properties:
      entity.dynamicLocation.flags:
        user: root
```

**Comment by aledsage** [comment]
Marked this as a bug because the default `.bom` file that ships with clocker is broken without this fix.

**Comment by grkvlt** [comment]
Fixed by #260
